### DISCUSSION TITLE
How to view Deno KV data locally?

### DISCUSSION THREAD

**Discussion by adamzerner** [denoland/deno#20580] [Q&A]
I'm working on an app and did something to get the following code to run:
```
kv.set(["users", email], {
  email,
  hashedPassword,
  shouldReceiveEmailUpdates,
});
```
Now I want to see the values in some sort of GUI, edit them, delete them, and create new values. How can I do that?

**Comment by EdamAme-x** [comment]
In other words, do you want me to tell you what is necessary to operate KV data on the GUI?

**Reply by EdamAme-x** [reply]
I believe that using the CLI, you can establish a connection to the mentioned thing.

Although it may require some effort, I think you can also manipulate it by awaiting kv.get(['key']);, considering the deployment destination.

Doesn't that sound reasonable?

The linked article is written in Japanese, but it might provide useful information. https://qiita.com/t_o_d/items/cbd06a9f16875eac53ba

**Comment by ajinkyashukla** [comment]
The name of the file on local is called `kv.sqlite3`. On Linux and Mac OS, you can find it using `locate kv.sqlite3`.

**Comment by AbdulrhmanGoni** [comment]
> I'm working on an app and did something to get the following code to run:
> 
> ```
> kv.set(["users", email], {
>   email,
>   hashedPassword,
>   shouldReceiveEmailUpdates,
> });
> ```
> 
> Now I want to see the values in some sort of GUI, edit them, delete them, and create new values. How can I do that?

Two years later, I ran into the same problem and ended up building a desktop app to manage Deno KV databases (browse, add, edit and delete entries through a GUI). 

If you’re still interested or if others land here, here’s a solution you can try: [denokv-gui-client](https://abdulrhmangoni.github.io/denokv-gui-client/).

**Comment by ooker777** [comment]
From [Cloud - Deno documentation \| Deno Docs](https://docs.deno.com/api/deno/cloud/#Deno.openKv):
> When no path is provided, the database will be opened in a default path for the current script. This location is persistent across script runs and is keyed on the origin storage key (the same key that is used to determine `localStorage` persistence). More information about the origin storage key can be found in the Deno Manual.

Use `deno info` to get your `DENO_DIR` env. On Windows it's probably on `C:\Users\[user name]\AppData\Local\deno\location_data\[hash]\local_storage`

**Comment by gitsult4n** [comment]
Deno KV's local file is SQLite-backed, but values are V8-serialized, so it isn't readable in a plain SQLite browser. Easiest option: `kv-toolbox` (`jsr:@kitsonk/kv-toolbox`) ships an explorer route you can mount in your app that gives a web UI to browse, edit, and delete keys. For a deployed database (Deno Deploy), there's already a built-in KV data browser in the dashboard. There isn't an official standalone GUI/CLI for local KV yet.
